src/compiler/checker.ts

```diff
@@ -27612,12 +27612,16 @@ namespace ts {
                 if (!isErrorType(intrinsicClassAttribs)) {
                     const typeParams = getLocalTypeParametersOfClassOrInterfaceOrTypeAlias(intrinsicClassAttribs.symbol);
                     const hostClassType = getReturnTypeOfSignature(sig);
-                    apparentAttributesType = intersectTypes(
-                        typeParams
-                            ? createTypeReference(intrinsicClassAttribs as GenericType, fillMissingTypeArguments([hostClassType], typeParams, getMinTypeArgumentCount(typeParams), isInJSFile(context)))
-                            : intrinsicClassAttribs,
-                        apparentAttributesType
-                    );
+                    let libraryManagedAttributeType: Type | undefined;
+                    if (typeParams && typeParams.length === 1) {
+                        const inferredArgs = fillMissingTypeArguments([hostClassType], typeParams, getMinTypeArgumentCount(typeParams), isInJSFile(context));

```

Comment:
oh, since I checked the typeParams length, I should not call `fillMissingTypeArguments`.

Which is preferred? To limit the generics number of JSX.IntrinsicClassAttributes = 1, or allow any argument count and call `fillMissingTypeArguments` to fill it?

@weswigham

Reply:
I usually err on the side of being permissive, since we've never surfaced real errors for these intrinsic types - so allowing too many arguments, but only providing the first and defaulting/constraint'ing the rest. Not much reasoning for either way, though. It's essentially a quiet error case.

Reply:
Actually, they're not intrinsic types, they're installed from npm. That's also true for libes6.d.ts etc, you can use `nolib` and provide you're own `Promise<T>` type. Maybe there's more incorrect cast in the checker assuming the shape of the library types and might crash under some edge cases.

Reply:
Most use the `getGlobalType` function which issues an error if arity doesn't match what we expect. The JSX types have just always been a bit more lax.

Reply:
I opened https://github.com/microsoft/TypeScript/pull/50271 to clean up the code.